ha/setup/ha_setup.py

```diff
@@ -0,0 +1,315 @@
+#!/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import sys
+import errno
+import argparse
+import inspect
+import traceback
+import os
+import shutil
+
+from cortx.utils.conf_store import Conf
+from cortx.utils.log import Log
+from cortx.utils.validator.v_pkg import PkgV
+from cortx.utils.security.cipher import Cipher
+from ha.execute import SimpleCommand
+from ha import const
+
+class Cmd:
+    """
+    Setup Command
+    This class provides methods for parsing arguments.
+    """
+    _index = "conf"
+
+    def __init__(self, args: dict):
```

Comment:
Looks like parse_args() takes dictionary arguments.
Moreover this template was shared by architecture team and did not face any issues while testing. Please let me know if you see any issues so that we need to change.